test/com/amazon/ion/streaming/ReaderTest.java

```diff
@@ -243,13 +243,10 @@ public void testReadingDecimalAsLong()
         assertEquals(readLong("9223372036854775807."), readLong("9223372036854775807")); // Max long
     }
 
-    @Test
+    @Test(expected = IllegalStateException.class)
     public void testReadingDecimalAsBigInteger()
     {
-        assertEquals(readBigInteger("0."), readBigInteger("0"));
-        assertEquals(readBigInteger("null.decimal"), readBigInteger("null.int"));
-        assertEquals(readBigInteger("9223372036854775807."), readBigInteger("9223372036854775807")); // Max long
-        assertEquals(readBigInteger("2d24"), readBigInteger("2000000000000000000000000"));
+        readBigInteger("0.");
```

Comment:
Suggestion:

```suggestion
    @Test
    public void testReadingDecimalAsBigInteger() {
        assertThrows(IllegalStateException.class, () -> readBigInteger("0."));
    }
```

With the attendant import up above:
```java
import static org.junit.jupiter.api.Assertions.assertThrows;
```

Reply:
Done.